internal/service/flexcluster/model.go

```diff
@@ -1,22 +1,141 @@
+		return nil, diags
+	}
+	return &admin.FlexClusterDescriptionCreate20250101{
+		Name: plan.Name.ValueString(),
+		ProviderSettings: admin.FlexProviderSettingsCreate20250101{
+			BackingProviderName: providerSettings.BackingProviderName.ValueString(),
+			RegionName:          providerSettings.RegionName.ValueString(),
+		},
+		TerminationProtectionEnabled: plan.TerminationProtectionEnabled.ValueBoolPointer(),
+		Tags:                         newResourceTags(ctx, plan.Tags),
+	}, nil
+}
+
+func NewAtlasUpdateReq(ctx context.Context, plan *TFModel) (*admin.FlexClusterDescription20250101, diag.Diagnostics) {
+	updateRequest := &admin.FlexClusterDescription20250101{
+		TerminationProtectionEnabled: plan.TerminationProtectionEnabled.ValueBoolPointer(),
+		Tags:                         newResourceTags(ctx, plan.Tags),
+	}
+
+	return updateRequest, nil
+}
+
+func ConvertBackupSettingsToTF(ctx context.Context, backupSettings *admin.FlexBackupSettings20250101) (*types.Object, diag.Diagnostics) {
+	if backupSettings == nil {
+		backupSettingsTF := types.ObjectNull(BackupSettingsType.AttributeTypes())
+		return &backupSettingsTF, nil
+	}
+
+	backupSettingsTF := &TFBackupSettings{
+		Enabled: types.BoolPointerValue(backupSettings.Enabled),
+	}
+	backupSettingsObject, diags := types.ObjectValueFrom(ctx, BackupSettingsType.AttributeTypes(), backupSettingsTF)
+	if diags.HasError() {
+		return nil, diags
+	}
+	return &backupSettingsObject, nil
+}
+
+func ConvertConnectionStringsToTF(ctx context.Context, connectionStrings *admin.FlexConnectionStrings20250101) (*types.Object, diag.Diagnostics) {
+	if connectionStrings == nil {
+		connectionStringsTF := types.ObjectNull(ConnectionStringsType.AttributeTypes())
+		return &connectionStringsTF, nil
+	}
+
+	connectionStringsTF := &TFConnectionStrings{
+		Standard:    types.StringPointerValue(connectionStrings.Standard),
+		StandardSrv: types.StringPointerValue(connectionStrings.StandardSrv),
+	}
+	connectionStringsObject, diags := types.ObjectValueFrom(ctx, ConnectionStringsType.AttributeTypes(), connectionStringsTF)
+	if diags.HasError() {
+		return nil, diags
+	}
+	return &connectionStringsObject, nil
+}
+
+func ConvertProviderSettingsToTF(ctx context.Context, providerSettings admin.FlexProviderSettings20250101) (*types.Object, diag.Diagnostics) {
+	providerSettingsTF := &TFProviderSettings{
+		ProviderName:        types.StringPointerValue(providerSettings.ProviderName),
+		RegionName:          types.StringPointerValue(providerSettings.RegionName),
+		BackingProviderName: types.StringPointerValue(providerSettings.BackingProviderName),
+		DiskSizeGb:          types.Float64PointerValue(providerSettings.DiskSizeGB),
+	}
+	providerSettingsObject, diags := types.ObjectValueFrom(ctx, ProviderSettingsType.AttributeTypes(), providerSettingsTF)
+	if diags.HasError() {
+		return nil, diags
+	}
+	return &providerSettingsObject, nil
+}
+
+func newTFTags(tags *[]admin.ResourceTag) basetypes.MapValue {
```

Comment:
tags conversion logic seems identical to the one we have in [project resource](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/project/model_project.go#L212-L234), could be a good opportunity to extract common code

Reply:
Will do that in a followup ticket so that we focus this PR in flex_clusters